health.go

```diff
@@ -97,17 +98,24 @@ func (h *health) scheduleCheck(task *checkTask, cfg *Config) {
 		if !h.runCheckOrStop(task, time.After(cfg.InitialDelay)) {
 			return
 		}
-
+		h.reportResults()
```

Comment:
It's not immediately obvious why results needs to be reported twice. Maybe add some comments clarifying?

Reply:
This preserves the original behaviour that was here. Each check we run - we report the results - now using the `HealthListener`

Reply:
We need to zero the metrics, and that's why you need an extra event instead of the code we had initially. It won't hurt to have a comment there indeed, cause you asked about this too

Reply:
It's not for zeroing the metrics, the zero happens in `OnCheckRegister` event. 
This report is because we call `runCheckOrStop` earlier in the flow.